src/app/app.component.ts

```diff
@@ -12,14 +12,18 @@ export class AppComponent {
     alert("I did stuff");
   }
 
-  doOtherStuff() {
-    try {
-      for (let i = 0; i < 50; i++) {
-        console.log('looping for no good reason');
-      }
-      alert("I did other stuff");
-    } catch (e) {
-      throw e;
+  public doOtherStuff() {
+    var a = [1, 2, 3];
+    var b = a.reverse(); // Noncompliant
+  }
+
+  public foo(a: boolean, b: boolean) {
+    if (a) {
+      console.log('a!');
+    } if (b) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Move this "if" to a new line or add the missing "else". ([typescript:S3972](http://localhost:9076/coding_rules#rule_key=typescript%3AS3972))

[See it in SonarQube](http://localhost:9076/project/issues?id=org.sonarqube%3Aangular-sample&pullRequest=1&issues=AWmQHKGAZQKnx6JGnxzl&open=AWmQHKGAZQKnx6JGnxzl)

[ISSUE_KEY]: AWmQHKGAZQKnx6JGnxzl